File: HelperJobby/DTOs/Job/CreateUpdateSalaryDTO.cs
using ApplicationDomain.Enums;

namespace HelperJobby.DTOs.Job;

public class CreateUpdateSalaryDTO
{
    public ShowPayByOptions ShowPayByOption { get; set; }
    public decimal MinimalAmount { get; set; }
    public decimal? MaximalAmount { get; set; }
    public SalaryRates SalaryRate { get; set; }
}

File: HelperJobby/client-app/src/DTOs/jobRelatetedDTOs/CreateUpdateSalaryDTO.ts
import {ShowPayByOptions} from "../../enums/modelDataEnums/ShowPayByOptions";
import {SalaryRates} from "../../enums/modelDataEnums/SalaryRates";

export interface CreateUpdateSalaryDTO {
    showPayByOption: ShowPayByOptions;
    minimalAmount: number;
    maximalAmount?: number;
    meetsMinSalaryRequirement: boolean;
    salaryRate: SalaryRates;
}